pkg/agent/params.go

```diff
@@ -147,6 +151,15 @@ func getParameters(cs *api.ContainerService, generatorCode string, bakerVersion
 	return parametersMap
 }
 
+func assignKubernetesParametersFromAgentProfile(profile *api.AgentPoolProfile, parametersMap paramsMap,
+	cloudSpecConfig api.AzureEnvironmentSpecConfig, generatorCode string) {
+	if profile.KubernetesConfig != nil {
+		// override containerRuntime parameter value if specified in AgentPoolProfile
+		// this allows for heteregenous clusters
+		addValue(parametersMap, "containerRuntime", profile.KubernetesConfig.ContainerRuntime)
```

Comment:
what is the desired behavior if profile.KubernetesConfig.ContainerRuntime is empty? Should we skip it ?

Reply:
good point - yea we should skip it. I'll add the check in